swarm/storage/ldbstore.go

```diff
@@ -753,6 +815,12 @@ func (s *LDBStore) setCapacity(c uint64) {
 
 func (s *LDBStore) Close() {
 	close(s.quit)
+	s.lock.Lock()
+	s.closed = true
+	s.lock.Unlock()
+	// force writing out current batch
+	s.writeCurrentBatch()
```

Comment:
shouldn't this be before unlocking?

Reply:
`writeCurrentBatch` itself starts with getting a lock, so if we would call it before unlocking, that would cause a deadlock.